utils/api/schemas/pub.ts

```diff
@@ -246,14 +251,7 @@ export const importCreateParams = optionalPubCreateParamSchema
 export type ImportCreatePubParams = (typeof importCreateParams)['_input'];
 
 export const base = z.object({
-	files: z.union([
-		z.array(
-			z
-				.tuple([z.custom<Blob>(), z.string({ description: 'filename' })])
-				.openapi({ title: 'Blob + filename (Node 18)' }),
-		),
-		z.custom<File[]>().openapi({ title: 'File (Node 20+, browser)' }),
-	]),
+	files: z.union([z.array(fileSchema), fileSchema]),
```

Comment:
this is kind of a significant change, and is related to something i changed in our fork of `ts-rest`, which i'll explain here.

`ts-rest` does not do arrays well. if it encounters an array in a field in an object that gets converted to formdata, it just appends the whole array, which often leads you with `"['object Object', 'object Object']"` as the field, which is silly.

before, I would allow users to pass either a `File` (which is not available in node 18), or `[Blob, string]`. I would then manually, in the `sdk`, create methods that merge these things and do something like

```ts
upload({ files }) {
  const form = new FormData()
  if(Array.isArray(files)){
       files.forEach(([blob, filename]) => {
          form.append('files', blob, filename)
       })
  }
// ...
}
```

which makes formdata sort of manually create a file. Kind of weird but whatever.

I decided to get rid of those manual methods in the `sdk`, because, as we were forking `ts-rest` anyway, might as well have it do what we want it to do.

See this diff for what i changed

https://github.com/pubpub/ts-rest/compare/b533bdd39b0b177733e01ffcfc4de8aa3518d077...fe7c011b7dcc7af6f11a3785449695c6ae72e7d7#diff-aa23f3f5a1b6dab4c323411e117b8624de01d63a532d78e5138363cde570eba3
  
Long story short, allowing for both `File` and a `Blob`/`filename` pair, and arrays of those, is way cleaner if the `Blob`/`filename` pair is an object.

so to upload files, you can now do

```ts
await pubpub.upload.file({
     files: [{ blob: new Blob(['hey'], 'text/plain'), filename: 'hey.txt' }]
})
```

Neat!